syntax = "proto3";
package angr;

/* TODO (discuss):
 *  track POI source (user/machine/tool w/ IP, etc)
 *  message CodeRef (file, addr/line, etc)
 * 
 */

message HumanPOI {
    string tool = 1;    // the source tool (e.g., "angr-management")
    int64 timestamp = 2; // see: https://stackoverflow.com/questions/3574716/date-and-time-type-for-use-with-protobuf
    string source = 3; // IP, username, hostname, etc
    string file = 4;   // binary, source file

    int64 code_location = 5;

    enum LocType {
        SOURCE = 0;         // source code line number
        RAW_BIN_ADDR = 1;   // arbitrary address in a binary (sure you don't mean INST_ADDR?)
        BLK_ADDR = 2;       // basic block (i.e., not instruction at the address)
        FUNC_ADDR = 3;      // function (i.e., not instruction at the address)
        INST_ADDR = 4;      // instruction/operands at the address
    };

    LocType loc_type = 6;
}

message POIList {
    repeated HumanPOI pois = 1;
}

